### ISSUE TITLE
field name with . inside it

### ISSUE THREAD

**Issue by 0xff00ff** [simonnilsson/ask#3]
Is there any plan to accept fields which has a dot inside their names?
ex:
```
{
    "field": {
        "sub.field": "needed value"
    }
}
```

**Comment by 0xff00ff** [comment]
short research shows that you can just replace strings.Split into something like this: https://play.golang.com/p/tzV7sfeFO8U and it will work well

**Comment by simonnilsson** [comment]
Hi @0xff00ff , a counterproposal.

What about keeping For/Path as is and adding two new varadic functions that can take the path already split, you could then use your own split method or just pass them as arguments. Example

```go
ask.ForArgs(object, customSplit("field.sub\\.field")...)
```

or simply

```go
ask.ForArgs(object, "field", "sub.field")
```

This function could also take integers allowing these two calls to give the same result:

```go
ask.For(object, "a[0].b.c")
ask.ForArgs(object, "a", 0, "b", "c")
```

**Comment by 0xff00ff** [comment]
that would be a good improvement, thanks

**Comment by simonnilsson** [comment]
Implemented in v0.4.0.
